tombstone_gc.hh

```diff
@@ -28,23 +30,36 @@ class database;
 
 }
 
-class repair_history_map;
-using per_table_history_maps = std::unordered_map<table_id, seastar::lw_shared_ptr<repair_history_map>>;
+using reconcile_history_map = boost::icl::interval_map<dht::token, gc_clock::time_point, boost::icl::partial_absorber, std::less, boost::icl::inplace_max>;
+
+class per_table_history_maps {
+public:
+    std::unordered_map<table_id, seastar::lw_shared_ptr<reconcile_history_map>> _repair_maps;
+    seastar::lw_shared_ptr<reconcile_history_map> _group0_map;
+};
 
 class tombstone_gc_options;
 
 using gc_time_min_source = std::function<gc_clock::time_point(const table_id&)>;
 
 class tombstone_gc_state {
     gc_time_min_source _gc_min_source;
-    per_table_history_maps* _repair_history_maps;
+    per_table_history_maps* _reconcile_history_maps;
     gc_clock::time_point check_min(schema_ptr, gc_clock::time_point) const;
+
+    seastar::lw_shared_ptr<reconcile_history_map> get_repair_history_for_table(const table_id& id) const;
```

Comment:
repair -> reconcile?

Reply:
Here I'm not actually sure and kept the "repair" as these are really used just for the "repair" part (with the previous code for tombstone GC mode = "repair" and also together with using the "update_repair_time"). these were just moved to the private class section (as they are not being used outside of the class)

The new code for the group0 is using different methods ("group0_history*") as it is also using the separate map for storing the gc times for the whole group0 at once.